### ISSUE TITLE
history-graph (for sensors with UoM) doesn't show a friendly_name if the graph contains only ONE sensor

### ISSUE THREAD

**Issue by kristjanbjarni** [home-assistant/frontend#12316]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

history-graph only shows entity name and not friendly_name on a sensor, when there is exactly one sensor in the graph.

### Describe the behavior you expected

I would expect to see the friendly_name of the sensor to be displayed instead of the entity name, just like there is when there are two or more sensor in the graph.

### Steps to reproduce the issue

1. Add history-graph card
2. Add only one sensor to this card

### What version of Home Assistant Core has the issue?

core-2022.4.2

### In which browser are you experiencing the issue with?

Firefox 99.0

### Which operating system are you using to run this browser?

Windows 10

### State of relevant entities

```txt
State: 9.1
Attributes:
  state_class: measurement
  unit_of_measurement: °C
  device_class: temperature
  friendly_name: Temperature outside
```

### Problem-relevant frontend configuration

```yaml
type: history-graph
entities:
  - entity: sensor.sonoff_th16_2_temperature_2
hours_to_show: 48
refresh_interval: 0
```

**Comment by ildar170975** [comment]
> history-graph only shows entity name and not friendly_name on a sensor, when there is exactly one sensor in the graph.

If only one sensor is specified, then the `history-graph` does not show names at all:
```
type: history-graph
entities:
  - entity: sun.sun
hours_to_show: 24
```
![image](https://user-images.githubusercontent.com/71872483/164336493-81f79bbe-0f5d-45b0-9f26-fb255de8c3f6.png)

If two or more sensors are specified, then the card shows friendly names:
```
type: history-graph
entities:
  - entity: sun.sun
  - entity: person.ildar
hours_to_show: 48
```
![image](https://user-images.githubusercontent.com/71872483/164336675-f6af0b75-b995-4f4c-af81-9b10439eb9af.png)

This works as expected.

**Comment by kristjanbjarni** [comment]
I am referring to graphs with numerical sensor values:
```
type: history-graph
entities:
  - entity: sensor.sonoff_th16_2_temperature_2
```
![One_graph](https://user-images.githubusercontent.com/1550962/164351979-48b193f6-945a-4f54-9fa1-07aaeedab368.png)


```
type: history-graph
entities:
  - entity: sensor.sonoff_th16_2_temperature_2
  - entity: sensor.sonoff_th16_1_temperature
```
![Two_graphs](https://user-images.githubusercontent.com/1550962/164352002-4b73b106-3d25-488c-85e4-6ec0fb7d8207.png)

**Comment by ildar170975** [comment]
I see your point.
Here is my case:

One sensor, a pseudo-name is displayed.
Means - name converted automatically from entity_id.
![image](https://user-images.githubusercontent.com/71872483/164445209-95f62980-9bac-4f19-a65b-45647e672068.png)

Two sensors, entity_ids are displayed.
![image](https://user-images.githubusercontent.com/71872483/164445273-dd40fe05-ccba-4fa5-b886-704120396f24.png)

In both cases no "friendly_name" were specified for these template sensors.

**Comment by pearson** [comment]
@kristjanbjarni , thank you for reporting this. I just came here to make the same report, so you've saved me some work! 😄 

Hopefully this gets fixed soon.

**Comment by pearson** [comment]
Note that you can see this problem through the UI editor itself. If you have two entities listed, the friendly names are shown in the preview, but if you remove one of the entities the name for the remaining entity switches to the "pseudo name".

![Editor - both](https://user-images.githubusercontent.com/656880/165156285-3d14b1df-b6d9-41cc-bfa5-d7f82e8acbf5.png)
![Editor - one](https://user-images.githubusercontent.com/656880/165156294-121f47d5-ec3f-46bc-b8bc-83728a155654.png)

**Comment by ildar170975** [comment]
The test card (no `friendly_name` attributes are set for entities):
```
type: vertical-stack
cards:
  - type: history-graph
    entities:
      - entity: sensor.xiaomi_cg_1_co2
      - entity: sensor.xiaomi_cg_2_co2
  - type: history-graph
    entities:
      - entity: sensor.xiaomi_cg_1_co2
```
![image](https://user-images.githubusercontent.com/71872483/191351042-b58a7fd9-d04a-4158-ae81-14ed98341c94.png)
So [this issue](https://github.com/home-assistant/frontend/issues/12316#issuecomment-1105068184) is not observed.

Now on History page:
![image](https://user-images.githubusercontent.com/71872483/191351668-83316862-d456-4136-9156-0d512b3926cc.png)
![image](https://user-images.githubusercontent.com/71872483/191351617-92c156b2-d3b5-49b1-b019-27df25b9bb79.png)

----------------

Then let's check entities with `friendly_name` specified:
```
type: vertical-stack
cards:
  - type: history-graph
    entities:
      - entity: sensor.home_openweathermap_forecast_precipitation_probability
      - entity: sensor.village_openweathermap_forecast_precipitation_probability
  - type: history-graph
    entities:
      - entity: sensor.home_openweathermap_forecast_precipitation_probability
```
![image](https://user-images.githubusercontent.com/71872483/191352428-04979c59-a8b3-4409-b9fb-005c0a365a5a.png)

Same - in both cases `friendly_name` is displayed.

**Seems that the issue should be CLOSED.**

**Comment by kristjanbjarni** [comment]
Yes I can confirm, this seems to be fixed in latest version 2022.9.5

**Comment by pearson** [comment]
Yes, this is now fixed. Well done, to whomever did it!
